### PR TITLE
[PluggableDevice] Add DEVICE_DEFAULT registration for int32 Const

### PR DESCRIPTION
**Pull request by PatriceVignola**
Add DEVICE_DEFAULT registration for int32 Const so that pluggable devices don't have to register it themselves.

### CODE REVIEW

**Review by sanjoy — CHANGES_REQUESTED** [review]
Can we remove the registration for `DEVICE_GPU`?

CC @penpornk

**Comment by PatriceVignola** [comment]
> Can we remove the registration for `DEVICE_GPU`?
> 
> CC @penpornk

Thanks for the review! I removed the GPU registration.

File: tensorflow/core/kernels/host_constant_op.cc [hunk 1]
```diff
@@ -43,17 +43,14 @@ void _HostConstantOp::Compute(OpKernelContext* ctx) {
   ctx->set_output(0, tensor_);
 }
 
-#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
 // A special GPU kernel for int32.
```
Comment (sanjoy) [inline tensorflow/core/kernels/host_constant_op.cc:46]: Update comment?

**Review by sanjoy — CHANGES_REQUESTED** [review]

**Review by sanjoy — APPROVED** [review]

**Comment by PatriceVignola** [comment]
@sanjoy Do I need to do anything to satisfy the import/copybara check?

**Comment by sanjoy** [comment]
@rthadur Do you know what's going on with the import/copybara failure?

**Comment by rthadur** [comment]
@sanjoy sorry I missed it

**Pull request merged**